src/main/java/winstone/HostConfiguration.java

```diff
@@ -128,27 +126,24 @@ private void loadBuiltinMimeTypes() {
      * @param webAppName
      *      Unique name given to the access logger.
      */
-    private Handler configureAccessLog(Handler handler, String webAppName) {
+    private RequestLog configureAccessLog(String webAppName) {
         try {
             Class<? extends RequestLog> loggerClass =
                     Option.ACCESS_LOGGER_CLASSNAME.get(args, RequestLog.class, commonLibCL);
             if (loggerClass != null) {
                 // Build the realm
                 Constructor<? extends RequestLog> loggerConstr = loggerClass.getConstructor(String.class, Map.class);
-                RequestLogHandler rlh = new RequestLogHandler();
-                rlh.setHandler(handler);
-                rlh.setRequestLog(loggerConstr.newInstance(webAppName, args));
-                return rlh;
+                return loggerConstr.newInstance(webAppName, args);
             } else {
                 Logger.log(Level.FINER, Launcher.RESOURCES, "WebAppConfig.LoggerDisabled");
             }
         } catch (Throwable err) {
             Logger.log(Level.SEVERE, Launcher.RESOURCES, "WebAppConfig.LoggerError", "", err);
         }
-        return handler;
+        return null;
     }
 
-    private WebAppContext create(File app, String prefix) {
+    private WebAppContext create(File app, String prefix, Server server) {
```

Comment:
There is already a field named `Server server`. Shadowing this field in an argument variable is a confusing practice and Checkstyle has a [`HiddenField`](https://checkstyle.sourceforge.io/checks/coding/hiddenfield.html#HiddenField) check that complains about this. I recommend simply reverting this hunk so that the field is used rather than the parameter:

```suggestion
    private WebAppContext create(File app, String prefix) {
```